libblkid/src/probe.c

```diff
@@ -737,6 +778,7 @@ const unsigned char *blkid_probe_get_buffer(blkid_probe pr, uint64_t off, uint64
 		if (!bf)
 			return NULL;
 
+		mark_prunable_buffers(pr, bf);
```

Comment:
Maybe we can also add to the `struct blkid_probe` 
- `nprunables` counter to do nothing in blkid_probe_prune_buffers() if the counter is zero
- `furthermost_offset` to do nothing in mark_prunable_buffers() if the current read() is after the previous read calls.
Just idea; maybe it's premature optimization ;-)

Reply:
Feels a bit like premature optimization :-)

Instead of adding `nprunables` we could also move the prunable buffers to a dedicated second list.

Reply:
Yeah, a second list sounds good, this solution is with minimal overhead ... (I remember this from PostgreSQL; there was a list of unused memory areas to recycle.)

Reply:
Second list is implemented